components/misc/fixedpoint.cpp

```diff
@@ -108,12 +114,16 @@ double FixedPoint::toDouble() const
 std::string FixedPoint::str() const
 {
     std::stringstream ss;
-    ss << intPart();
```

Comment:
This didn't work for negatives if intPart is zero.
Similar for from string constructor.
`fractionPart` above dropped the sign, and `round` didn't work for negatives